sdk/src/account_meta.rs

```diff
@@ -1,79 +1,193 @@
+//         let address_merkle_context =
+//             address_merkle_context.map(|ctx| pack_address_merkle_context(ctx, remaining_accounts));
+//         Ok(Self {
+//             lamports: None,
+//             address: None,
+//             data: None,
+//             merkle_context: None,
+//             merkle_tree_root_index: None,
+//             output_merkle_tree_index: Some(output_merkle_tree_index),
+//             address_merkle_context,
+//             address_merkle_tree_root_index,
+//             read_only: false,
+//         })
+//     }
+
+//     #[allow(clippy::too_many_arguments)]
+//     pub fn new_mut(
+//         compressed_account: &CompressedAccountWithMerkleContext,
+//         merkle_tree_root_index: u16,
+//         output_merkle_tree: &Pubkey,
+//         accounts: &mut LightInstructionAccounts,
+//     ) -> Self {
+//         let merkle_context = pack_merkle_context(&compressed_account.merkle_context, accounts);
+
+//         // If no output Merkle tree was specified, use the one used for the
+//         // input account.
+//         let output_merkle_tree_index = accounts.insert_or_get(*output_merkle_tree);
+
+//         Self {
+//             lamports: Some(compressed_account.compressed_account.lamports),
+//             address: compressed_account.compressed_account.address,
+//             data: compressed_account
+//                 .compressed_account
+//                 .data
+//                 .as_ref()
+//                 .map(|data| data.data.clone()),
+//             merkle_context: Some(merkle_context),
+//             merkle_tree_root_index: Some(merkle_tree_root_index),
+//             output_merkle_tree_index: Some(output_merkle_tree_index),
+//             address_merkle_context: None,
+//             address_merkle_tree_root_index: None,
+//             read_only: false,
+//         }
+//     }
+
+//     pub fn new_close(
+//         compressed_account: &CompressedAccountWithMerkleContext,
+//         merkle_tree_root_index: u16,
+//         accounts: &mut LightInstructionAccounts,
+//     ) -> Self {
+//         let merkle_context = pack_merkle_context(&compressed_account.merkle_context, accounts);
+//         Self {
+//             lamports: Some(compressed_account.compressed_account.lamports),
+//             address: compressed_account.compressed_account.address,
+//             data: compressed_account
+//                 .compressed_account
+//                 .data
+//                 .as_ref()
+//                 .map(|data| data.data.clone()),
+//             merkle_context: Some(merkle_context),
+//             merkle_tree_root_index: Some(merkle_tree_root_index),
+//             output_merkle_tree_index: None,
+//             address_merkle_context: None,
+//             address_merkle_tree_root_index: None,
+//             read_only: false,
+//         }
+//     }
+// }
+
+#[derive(Debug, Clone, PartialEq)]
 pub struct LightAccountMeta {
```

Comment:
should we add a type field (enum Init, Mut, ReadOnly, Close) which is used to pack the AccountMeta?